router/core/src/main/scala/io/buoyant/router/DiscardingFactoryToService.scala

```diff
@@ -0,0 +1,99 @@
+package io.buoyant.router
+
+import com.twitter.finagle._
+import com.twitter.util.{Future, Time}
+import io.buoyant.router.DiscardingFactoryToService.RequestDiscarder
+
+/**
+ * Turns a [[com.twitter.finagle.ServiceFactory]] into a
+ * [[com.twitter.finagle.Service]] which acquires a new service for
+ * each request. Additionally uses a [[RequestDiscarder]] to discard
+ * the request and release any resources associated with it.
+ */
+class DiscardingFactoryToService[Req, Rep](requestDiscarder: RequestDiscarder[Req], factory: ServiceFactory[Req, Rep]) extends Service[Req, Rep] {
+
+  def apply(request: Req): Future[Rep] =
+    factory().onFailure(_ => requestDiscarder.discard(request)).flatMap { service =>
+      service(request).ensure {
+        service.close()
+        ()
+      }
+    }
+
+  override def close(deadline: Time): Future[Unit] = factory.close(deadline)
+  override def status: Status = factory.status
+}
+
+object DiscardingFactoryToService {
+  val role = Stack.Role("DiscardingFactoryToService")
+
+  case class Enabled(enabled: Boolean) {
+    def mk(): (Enabled, Stack.Param[Enabled]) =
+      (this, Enabled.param)
+  }
+  object Enabled {
+    implicit val param = Stack.Param(Enabled(false))
+  }
+
+  case class RequestDiscarder[Req](discard: Req => Unit)
+  object RequestDiscarder {
+    private val _param = new Stack.Param[RequestDiscarder[Any]] {
+      val default = RequestDiscarder(_ => ()) // default is noop
+    }
+    def param[Req] = _param.asInstanceOf[Stack.Param[RequestDiscarder[Req]]]
+  }
+
+  /**
+   * Creates a [[com.twitter.finagle.Stackable]]
+   * [[DiscardingFactoryToService]]. This makes per-request service acquisition
+   * part of the stack so it can be wrapped by filters such as tracing.
+   */
+  def module[Req, Rep]: Stackable[ServiceFactory[Req, Rep]] = {
```

Comment:
I don't think this module is ever used so we can omit defining it.  In `Router` it looks like we just use DiscardingFactoryToService directly.